### PR TITLE
Only log internal server errors, don’t send details to the API user.

### LINKED ISSUE #573: Internal Server Errors should be logged at the ERROR level and the API response given to the client should not contain the details.

### PR DESCRIPTION
**Pull request by chirino**
Fixes #573

### CODE REVIEW

File: modules/fundamental/src/error.rs [hunk 1]
```diff
@@ -51,32 +51,53 @@ impl ResponseError for Error {
...
-            Error::Data(msg) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Data-model corruption", msg)),
+
+            // The following match arms handle internal system errors that are not expected to occur.
+            // They are logged and a generic error response is returned to avoid leaking
+            // internal state to end users.
+            Self::Actix(err) => {
+                log::error!("System Actix: {}", err);
+                HttpResponse::InternalServerError()
+                    .json(ErrorInformation::new("Internal Server Error", ""))
```
Comment (jcrossley3) [inline modules/fundamental/src/error.rs:74]: The json isn't really adding anything, so I think you can just...
```suggestion
                HttpResponse::InternalServerError().finish()
```

Comment (chirino) [reply modules/fundamental/src/error.rs:74]: I do tend to like APIs that have consistent response bodies on all errors, including 500s.

File: modules/fundamental/src/error.rs [hunk 2]
```diff
@@ -51,32 +51,53 @@ impl ResponseError for Error {
...
                 HttpResponse::BadRequest().json(ErrorInformation::new("Storage Key", err))
             }
-            Error::Data(msg) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Data-model corruption", msg)),
+
+            // The following match arms handle internal system errors that are not expected to occur.
+            // They are logged and a generic error response is returned to avoid leaking
+            // internal state to end users.
+            Self::Actix(err) => {
+                log::error!("System Actix: {}", err);
```
Comment (ctron) [inline modules/fundamental/src/error.rs:72]: I don't really like the pattern of adding a custom log message. IIRC we are using `thiserror` and so the error message should already be handled by the `Display` trait implementation. So a simple `log::error!("{err}")` should be sufficient. And given the fact that all match patterns would then have the same implementation, I think we can simplify this to just:

```rust
err => {
  log::error!("{err}");
  HttpResponse::InternalServerError()
    .json(ErrorInformation::new("Internal Server Error", ""))
}
```

Comment (chirino) [reply modules/fundamental/src/error.rs:72]: Sounds good.

**Review by ctron — APPROVED** [review]

**Pull request merged**